mlapi/facet_sense_api.py

```diff
@@ -0,0 +1,23 @@
+import requests
+from requests.auth import HTTPBasicAuth
+from definitions import Definitions
+from pathlib import Path
+import json
+
+
+class FacetSenseApi(object):
+    API_URL = "https://whispersherbrooke2018.cloud.coveo.com/predict"
+    QUERY = "{{\"q\": \"{text}\"}}"
+    CREDENTIALS_PATH = Path(Definitions.ROOT_DIR + "/appsettings.json")
+
+    def __init__(self):
+        file = open(FacetSenseApi.CREDENTIALS_PATH, 'r')
```

Comment:
Using the with keyword makes sure we always close the file without having, so you could delete the file.close() too

Example from WHisperAutomaticTests
https://github.com/CoveoWhisper/WhisperAutomaticTest/blob/3f86e2a785d8bfc5f891847c0bafd57246b23cc3/whisper_automatic_test/scenario_reader.py#L8

Reply:
good to know!